### PR TITLE
OpenAPI specifications for get source

### PR DESCRIPTION
**Pull request by rubhanazeem**
Document OpenAPI specifications for `/source` to get the list of projects or the list of deleted projects

### CODE REVIEW

**Comment by rubhanazeem** [comment]
[Sources - Projects (L)](https://trello.com/c/MSXY9t4w/1986-sources-projects-l)

File: src/api/public/apidocs-new/OBS-v2.10.50.yaml [hunk 1]
```diff
@@ -255,7 +255,8 @@ paths:
     $ref: 'paths/source_orderkiwirepos.yaml'
 
   # Sources - Projects
-
+  /source:
+    $ref: 'paths/source_projects.yaml'
```
Comment (eduardoj) [inline src/api/public/apidocs-new/OBS-v2.10.50.yaml:259]:
```suggestion
    $ref: 'paths/source.yaml'
```
Let's keep the convention that every path file has the name of its path and arguments.

File: src/api/public/apidocs-new/paths/source_projects.yaml [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+            type: object
+            properties:
+              code:
+                type: string
+                xml:
+                  attribute: true
+              summary:
+                type: string
+            xml:
+              name: status
```
Comment (eduardoj) [inline src/api/public/apidocs-new/paths/source_projects.yaml:35]:
```suggestion
            $ref: '../schemas/api_response.yaml'
```
It should be possible to use the `api_response.yaml` schema here.

Comment (rubhanazeem) [reply src/api/public/apidocs-new/paths/source_projects.yaml:35]: `api_response.yaml` has more information than we need here. We just need code and summary

Comment (eduardoj) [reply src/api/public/apidocs-new/paths/source_projects.yaml]: The schema defines all possible fields and how they are arranged. But it doesn't mean that we have to use all the fields, in all the examples.

If in this case the example uses ony `code` and `summary`, there is no need to define another schema inline. When we don't assign values to the other fields, those fields are simply not shown.

Comment (rubhanazeem) [reply src/api/public/apidocs-new/paths/source_projects.yaml]: I see. Okay, I've updated it

File: src/api/public/apidocs-new/paths/source.yaml [hunk 3]
```diff
@@ -0,0 +1,43 @@
+get:
+  summary: List all the projects
+  description: |
+    Get a list of projects or a list of deleted projects.
```
Comment (hennevogel) [inline src/api/public/apidocs-new/paths/source.yaml:4]:
```suggestion
    Get a list of projects
```

File: src/api/public/apidocs-new/paths/source.yaml [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+  parameters:
+    - name: deleted
+      in: query
+      schema:
+        type: integer
+      description: 0 for existing projects and 1 for deleted projects.
```
Comment (hennevogel) [inline src/api/public/apidocs-new/paths/source.yaml:10]:
```suggestion
      description: Set to 1 to include deleted projects in the listing. This is available for Admins only.
```

File: src/api/public/apidocs-new/paths/source.yaml [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+      example: 0
+      required: false
+
+  responses:
```
Comment (hennevogel) [inline src/api/public/apidocs-new/paths/source.yaml:14]: There is one more

```
osc api /source/\?deleted=0 
Server returned an error: HTTP Error 403: Forbidden
only admins can see deleted projects
```

**Review by hennevogel — CHANGES_REQUESTED** [review]

File: src/api/public/apidocs-new/paths/source.yaml (hunk 5 above)
Comment (rubhanazeem) [reply src/api/public/apidocs-new/paths/source.yaml:14]: I've documented it

**Review by hennevogel — APPROVED** [review]

**Pull request merged**